subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/DelegatingDependencyMetadata.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.component.model;
+
+import org.gradle.api.artifacts.component.ComponentSelector;
+import org.gradle.api.capabilities.Capability;
+import org.gradle.api.internal.attributes.AttributesSchemaInternal;
+import org.gradle.api.internal.attributes.ImmutableAttributes;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A {@link DependencyMetadata} implementation which delegates all method calls to a provided {@code delegate}.
+ */
+public abstract class DelegatingDependencyMetadata implements DependencyMetadata {
```

Comment:
🧐 Should this be named `AbstractDelegatingDependencyMetadata` to indicate it's not meant to be used other than via extension?

Reply:
The `abstract` keyword enforces that. Prefixing any given abstract class with `Abstract` reminds me of prefixing every interface with `I`. IMO its unnecessary. I like prefixing classes with `Abstract` when the abstract implementation is one of those "implement some but not all of this interface" implementations.